### DISCUSSION TITLE
Roadmap to non-experimental 1.0?

### DISCUSSION THREAD

**Discussion by mseeley** [posva/unplugin-vue-router#342] [Questions and Help]
Typed routes is a massively helpful feature towards developer experience and early identification of coding errors. I thank all the maintainers for their effort.

I would like to share feedback and ask for clarification regarding the roadmap for this plugin.

- `vue-router@4.1.0` in July 2022 released [added typed route support](https://github.com/vuejs/router/releases/tag/v4.1.0) via this plugin.
  - There have been > 30 releases for `unplugin-vue-router` since July 2022.
  - Regardless the plugin remains "experimental" in this repo and within Nuxt.
  - What is 1.0? Is there a timeline?
- What considerations have been made to integrate typed route generation into `vue-router` itself?
  - I've found it troublesome to educate our team around _which_ `useRouter` should be imported in our projects (`#vue-router`, `vue-router`, `vue-router/auto`, `#imports`, etc...).
  - Personally, I'm feeling friction from:
    - lack of a single `useRouter` implementation and typing
    - unstable/experimental integration of typed routes our into applications
- Are there plans to separate the type safe routes into a standalone package/plugin, separate of the data loading and file-based routing?
  - Type safe routes, data loading, file-based routing, on the surface, read like 3 distinct `vue-router` plugins.

Is there a road to 1.0 that finalizes type safe routes independently of data loading and file-based routing?

Thank you.

**Comment by posva** [comment]
I'm glad you find this helpful and thanks for the feedback, I really appreciate it.
Currently, I plan to move the types to vue-router for better native integration and release as experimental there to gather feedback from the community. The goal is for it to become just a stable feature of vue router.
If that happens, it will be possible to split the plugin into 2 (file-based routing + data loaders). That being said, I plan to add other compilation-time improvements to vue-router through this plugin. They are all here right now because that's currently the only way to have them typed.

Regarding `useRouter()`, in vue apps, always import from `vue-router/auto` (until this becomes unnecessary and you can import from vue-router directly). In Nuxt, I'm not sure which one should be used, but I guess you don't import and use auto imports.

Regarding a timeline, I wish I could have one. The reality is life is rough for me right now, and I can't spend as much time as I want in the different parts of the Vue.js Ecosystem I maintain. I'm currently spending about 75% of my time in consulting + [Mastering Pinia](https://masteringpinia.com/) to pay the bills. I'm still trying to shift that by finding ways to get [sponsoring](https://github.com/sponsors/posva) from companies rather than individuals
